src/Video.tsx

```diff
@@ -0,0 +1,1992 @@
+} from 'react-native';
+
+import { SafeAreaView, useSafeAreaInsets } from 'react-native-safe-area-context';
+
+import ReactNativeBlobUtil from 'react-native-blob-util';
+import WebView, { WebViewMessageEvent, WebViewNavigation } from 'react-native-webview';
+import Orientation, {
+  LANDSCAPE_LEFT,
+  LANDSCAPE_RIGHT,
+  PORTRAIT,
+} from 'react-native-orientation-locker';
+import RNVideo, {
+  LoadError,
+  OnBufferData,
+  OnLoadData,
+  OnProgressData,
+  TextTrackType,
+} from 'react-native-video';
+import GoogleCast, {
+  CastButton,
+  MediaLoadRequest,
+  RemoteMediaClient,
+} from 'react-native-google-cast';
+import PrefersHomeIndicatorAutoHidden from 'react-native-home-indicator';
+
+import ActionModal from './ActionModal';
+import VideoTimer from './VideoTimer';
+import VideoSettings from './VideoSettings';
+import AnimatedCustomAlert from './AnimatedCustomAlert';
+import DoubleTapArea from './DoubleTapArea';
+import networkSpeedService from './services/networkSpeed.service';
+import LiveTimer from './LiveTimer';
+
+import { svgs } from './img/svgs';
+import { IS_IOS, IS_TABLET, PIX_R, getMP3Array } from './helper';
+import type { IContent, IMp3, IVpe } from './entity';
+import type ISvg from './img/ISvg';
+import Mp3Option from './Mp3Option';
+const { AirPlay, AirPlayButton, AirPlayListener } = require('react-native-airplay-ios');
+
+const iconStyle = { width: 40, height: 40, fill: 'white' };
+let playPressedFirstTime = true;
+let secondsPlayed = 0;
+let startPlaySec = 0;
+let endPlaySec = 0;
+let videoW: number;
+let videoH: number;
+let aCasting: boolean | undefined;
+let gCasting: boolean;
+let orientation: string;
+let offlinePath: string;
+let quality: string | number = 'Auto';
+
+interface IVideo {
+  onPlayerReady?: () => void;
+  youtubeId?: string;
+  content: IContent;
+  type: 'audio' | 'video';
+  toSupport: () => void;
+  onBack?: () => void;
+  connection: boolean | null;
+  goToNextLesson?: () => void;
+  goToPreviousLesson?: () => void;
+  disableNext?: boolean;
+  disablePrevious?: boolean;
+  autoPlay?: boolean;
+  live: boolean;
+  liveData?: {
+    isLive?: boolean;
+    live_event_end_time?: string;
+    live_event_start_time?: string;
+    live_event_end_time_in_timezone?: string;
+    live_event_start_time_in_timezone?: string;
+  };
+  startTime?: number;
+  endTime?: number;
+  onStart?: () => void;
+  quality?: number;
+  onEnd?: () => void;
+  onGCastingChange?: (casting: boolean) => void;
+  onACastingChange?: (casting?: boolean) => void;
+  gCasting?: boolean;
+  onRefresh: () => void;
+  showControls: boolean;
+  paused: boolean;
+  onFullscreen?: (isFullscreen: boolean) => void;
+  orientationIsLocked?: boolean;
+  repeat: boolean;
+  onUpdateVideoProgress?: (
+    videoId: number | string,
+    id: number,
+    lengthInSec: number,
+    currentTime: number,
+    secondsPlayed: number,
+    mediaCategory: string,
+    apiCallDelay?: number
+  ) => void;
+  listening?: boolean;
+  styles: {
```

Comment:
these are mainly just for colors, so it might be easier to just send primaryColor and theme.text.colors as a prop and use them everywhere where necessary